dace/sdfg.py

```diff
@@ -1828,9 +1832,8 @@ def apply_transformations(self,
         while applied:
             applied = False
             # Find and apply immediately
-            for match in opt.get_pattern_matches(
+            for sdfg, match in opt.get_pattern_matches(
                     strict=strict, patterns=patterns, states=states):
-                sdfg = self.sdfg_list[match.sdfg_id]
```

Comment:
I want to note that this is OK only when transforming the SDFG in-place, like we do through the command-line. The changes are completely unsuitable for other use-cases, e.g. DIODE.

Reply:
I tested DIODE and it works.  Also, this method (`sdfg.apply_transformations`) is always in-place.

Reply:
Did you test that you can go back, i.e. "undo"?

Reply:
yes, I did multiple undos and re-transformations too

Reply:
as a side-note, transformations don't really work on nested SDFGs in master. (just checked)

Reply:
Let's open another branch/PR to fix this then.